duckduckgo/chrome-zeroclickinfo
# async XMLHttpRequest

**Reviewer:**



## Description:


- The addon reviewer asked us to switch to async XMLHttpRequest. I updated load.js to use the async method and updated anywhere else in the extension that used the old method to include a call back now.

- We really want the default settings to be loaded before starting any of the other modules. I'm including it as a JS file in the manifest so we can be sure it has loaded first. 
https://github.com/duckduckgo/chrome-zeroclickinfo/compare/beta...jd/async-get?expand=1#diff-60e6512a6a0d9f504ae29abefe8cfc23R1
https://github.com/duckduckgo/chrome-zeroclickinfo/compare/beta...jd/async-get?expand=1#diff-4b1eb3dc48c4e16d49db5b42298fe654R37

- Unrelated but I fixed it in this PR. We're sending undef requests our atb.js backend. The legacy portion of our code is still handling this for now.
https://github.com/duckduckgo/chrome-zeroclickinfo/compare/beta...jd/async-get?expand=1#diff-cee2b51a0aa6dfee6ce4439cff8794eeR10

## Steps to test this PR:

easylist has parsed data
<img width="622" alt="screen shot 2017-07-25 at 9 05 35 am" src="https://user-images.githubusercontent.com/1882892/28579092-d8b99ac4-7118-11e7-88d2-8dfee262a51f.png">

httpsWhitelist exists
<img width="864" alt="screen shot 2017-07-25 at 9 05 47 am" src="https://user-images.githubusercontent.com/1882892/28579095-dab38dbc-7118-11e7-97ad-9edbaf87be9e.png">

Go to sites you typically use for testing the extension and make sure the number and companies blocked looks good.

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
